overlay_new.py

```diff
@@ -0,0 +1,197 @@
+    spacing = 10
+
+    power_tolerance = 0.05
+
+    def __init__(self, screen_w: int, screen_h: int):
+        self.power_coord = (screen_w // 2, screen_h - 30 - CentrePower.spacing - CentrePower.rec_power_height)
+        self.rec_power_coord = (screen_w // 2, screen_h - 30)
+
+    def draw_base(self, canvas: Canvas):
+        pass
+
+    def draw_data(self, canvas: Canvas, data: Data):
+        power = data["power"]
+        rec_power = data["rec_power"]
+
+        power_str = f"{power:.0f}w"
+        rec_power_str = f"{rec_power:.0f}w rec"
+
+        if rec_power == 0:
+            power_colour = Colour.black
+        else:
+            power_diff = abs(power / rec_power - 1)
+            # Colour.green is to light against the white fairing
+            dark_green = (0, 191, 0, 255)
+            power_colour = dark_green if power_diff <= CentrePower.power_tolerance else Colour.red
+
+        canvas.draw_text(power_str, self.power_coord, CentrePower.power_size, power_colour, "centre")
+        canvas.draw_text(rec_power_str, self.rec_power_coord, CentrePower.rec_power_size, Colour.black, "centre")
+
+class Message(Drawable):
+    """ A drawable which displays any existing messages at the top of the
+        screen.
+
+        Lines are wrapped appropriately to ensure the entire message is
+        visible. """
+
+    text_size = 1
+    spacing = 10
+    _, text_height = Canvas.get_text_dimensions("blah", text_size)
+    chars_per_line = 75
+
+    def draw_base(self, canvas: Canvas):
+        pass
+
+    def draw_data(self, canvas: Canvas, data: Data):
+        if not data.has_message():
+            return
+        message = data.get_message()
+        display_str = f"Message: {message}"
+
+        # Display each line of the message
+        line_y_coord = Message.spacing + Message.text_height
+        for line in wrap(display_str, Message.chars_per_line):
+            coord = (Message.spacing, line_y_coord)
+            canvas.draw_text(line, coord, Message.text_size)
+            line_y_coord += Message.text_height + Message.spacing
+
+class OverlayNew(Overlay):
+
+    def __init__(self, bike=None):
+        super(OverlayNew, self).__init__(bike)
+
+        self.start_time = time()
+
+        # Generate coordinates for each of the data fields in the bottom corners.
+        # Note that these coordinates are for the bottom left corner of each field.
+        spacing = 20
+        row_coords = [
+            self.height - (2 * spacing + DataField.height),
+            self.height - spacing,
+        ]
+        col_coords = [
+            spacing,
+            2 * spacing + DataField.width,
+            self.width - 2 * (spacing + DataField.width),
+            self.width - (spacing + DataField.width),
+        ]
+        # This lambda returns the coordinates of the data field in column x, row y
+        data_field_coord = lambda x, y: (col_coords[x], row_coords[y])
+
+        # Create all drawable overlay objects
+        self.drawables = [
```

Comment:
I have a strong urge to convert the other overlays to use components, so that we can have all the component handling stuff in `overlay.py`, and all we have to do in the implementations is populate self.components. It'll be soooo clean!